File: gfg/2024_8_august/python/23_left_view_of_binary_tree.py
# CODE

def LeftView(root):
    if root is None: 
        return []
    res = []

    def solve(root, lvl):
        if root is None: return
        if len(res) == lvl: res.append(root.data)
        solve(root.left, lvl + 1)
        solve(root.right, lvl + 1)
    
    solve(root, 0)
    return res

File: gfg/2024_8_august/c++/23_left_view_of_binary_tree.cpp
// TIME COMPEXITY
// O(n)

// SPACE COMPEXITY
// O(n)

// CODE

void solver(Node * root,vector<int> &ans, int lvl){
    if(root == NULL) return ;
    if(lvl == ans.size()){
        ans.push_back(root->data);
    }
    if(root->left){
        solver(root->left,ans,lvl+1);
    }
    if(root->right){
        solver(root->right,ans,lvl+1);
    }
}
vector<int> leftView(Node *root)
{
   vector<int> ans;
   solver(root,ans,0);
   return ans;
}